### PR TITLE
requeue when all addon agent manfiestworks are not deleted

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>

### CODE REVIEW

File: pkg/controller/upgrade/cleanup_controller.go [hunk 1]
```diff
@@ -221,58 +237,54 @@ func (r *ReconcileCleanup) cleanupDeprecatedRoleBinding(addon *addonv1alpha1.Man
...
-func (r *ReconcileCleanup) deleteManifestWork(clusterName, name string) error {
-	work := &manifestworkv1.ManifestWork{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{
-		Namespace: clusterName,
-		Name:      name,
-	}, work)
-	if errors.IsNotFound(err) {
-		return nil
+	if err := r.deleteManifestWork(namespace, manifestWorkName(namespace, klusterletAddonCRDs)); err != nil {
+		return true, err
```
Comment (elgnay) [inline pkg/controller/upgrade/cleanup_controller.go:284]: false?

File: pkg/controller/upgrade/cleanup_controller.go [hunk 2]
```diff
@@ -221,58 +237,54 @@ func (r *ReconcileCleanup) cleanupDeprecatedRoleBinding(addon *addonv1alpha1.Man
...
 	}
-	if err != nil {
-		return err
+	if err := r.deleteManifestWork(namespace, manifestWorkName(namespace, klusterletAddonOperator)); err != nil {
+		return true, err
```
Comment (elgnay) [inline pkg/controller/upgrade/cleanup_controller.go:287]: false?

File: pkg/controller/upgrade/cleanup_controller.go (hunk 1 above)
Comment (zhiweiyin318) [reply pkg/controller/upgrade/cleanup_controller.go]: done

File: pkg/controller/upgrade/cleanup_controller.go (hunk 2 above)
Comment (zhiweiyin318) [reply pkg/controller/upgrade/cleanup_controller.go]: done

**Pull request merged**